src/fsharp/fsi/fsimain.fs

```diff
@@ -316,7 +314,13 @@ let evaluateSession(argv: string[]) =
 let MainMain argv = 
     ignore argv
     let argv = System.Environment.GetCommandLineArgs()
-    use e = new SaveAndRestoreConsoleEncoding()
+    let savedOut = Console.Out
```

Comment:
You might as well remove SaveAndRestoreConsoleEncoding from fsharp.compiler.private, and update fsc to do the same thing as you do here.

Reply:
I'll probably do that.

I personally feel that the entire contents of `lib.fs` and `illib.fs` should be abstracted out into its own library (not: NOT a package) since there is duplication of its contents throughout the codebase.